pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1634,6 +1629,7 @@ protected void handleAck(CommandAck ack) {
         final boolean hasRequestId = ack.hasRequestId();
         final long requestId = hasRequestId ? ack.getRequestId() : 0;
         final long consumerId = ack.getConsumerId();
+        final CommandAck finalAck = getBrokerService().getInterceptor() != null ? new CommandAck().copyFrom(ack) : null;
```

Comment:
`getBrokerService().getInterceptor()` is never null. There's a `BrokerInterceptorDisabled` instance when it's not really used, see https://github.com/apache/pulsar/blob/82237d3684fe506bcb6426b3b23f413422e6e4fb/pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/BrokerInterceptors.java#L89-L93 . 

It would be good to optimize this case so that the copy of the CommandAck isn't done when the interceptor is disabled.

Reply:
Handled by #18997

Reply:
You are right, great catch.